File: benchmark/c/cpw/number_jump_required_given_length_reach_point_form_d_0_origin_2d_plane/number_jump_required_given_length_reach_point_form_d_0_origin_2d_plane.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int number_jump_required_given_length_reach_point_form_d_0_origin_2d_plane ( int a, int b, int d ) ;
int number_jump_required_given_length_reach_point_form_d_0_origin_2d_plane ( int a, int b, int d ) {
  int temp = a;
  a = min ( a, b );
  b = max ( temp, b );
  if ( d >= b ) return ( d + b - 1 ) / b;
  if ( d == 0 ) return 0;
  if ( d == a ) return 1;
  return 2;
}

File: translation_result/else_bms/fail/c/cpw/number_jump_required_given_length_reach_point_form_d_0_origin_2d_plane/gemini/CAPR/number_jump_required_given_length_reach_point_form_d_0_origin_2d_plane.rs

fn number_jump_required_given_length_reach_point_form_d_0_origin_2d_plane(a: i32, b: i32, d: i32) -> i32 {
    let temp = a;
    let a = a.min(b);
    let b = temp.max(b);
    if d >= b {
        (d + b - 1) / b
    } else if d == 0 {
        0
    } else if d == a {
        1
    } else {
        2
    }
}
